org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -390,17 +391,21 @@ public ICompilationUnit handleChanged(DidChangeTextDocumentParams params) {
 
 				Range range = changeEvent.getRange();
 				int length;
-
+				IDocument document = JsonRpcHelpers.toDocument(unit.getBuffer());
+				final int startOffset;
 				if (range != null) {
-					length = changeEvent.getRangeLength().intValue();
+					Position start = range.getStart();
+					Position end = range.getEnd();
+					int endOffset = JsonRpcHelpers.toOffset(document, end.getLine(), end.getCharacter());
+					startOffset = JsonRpcHelpers.toOffset(document, start.getLine(), start.getCharacter());
+					length = endOffset - startOffset;
```

Comment:
I think we can replace this with [`DiagnosticsHelper.getLength(ICompilationUnit, Range)`](https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/DiagnosticsHelper.java#L65-L69) in this particular case.

Reply:
Done ✔